Makefile

```diff
@@ -0,0 +1,119 @@
+
+PROJECT_NAME ?= andela-socials
+ORG_NAME ?= bench-projects
+REPO_NAME ?= andela-socials
+
+
+DOCKER_TEST_COMPOSE_FILE := docker/test/docker-compose.yml
+
+DOCKER_TEST_PROJECT = $(PROJECT_NAME)test
+
+TARGET_MAX_CHAR_NUM=10
+
+DOCKER_REGISTRY ?= gcr.io
+
+
+ifeq ($(DOCKER_REGISTRY), docker.io)
+	REPO_FILTER := $(ORG_NAME)/$(REPO_NAME)
+else
+	REPO_FILTER := $(DOCKER_REGISTRY)/$(ORG_NAME)/$(REPO_NAME)[^[:space:]|\$$]*
+endif
+
+.PHONY: help
+
+## Show help
+help:
+	@echo ''
+	@echo 'Usage:'
+	@echo '${YELLOW} make ${RESET} ${GREEN}<target> [options]${RESET}'
+	@echo ''
+	@echo 'Targets:'
+	@echo ''
+	@awk '/^[a-zA-Z\-\_0-9]+:/ { \
+		message = match(lastLine, /^## (.*)/); \
+		if (message) { \
+			command = substr($$1, 0, index($$1, ":")-1); \
+			message = substr(lastLine, RSTART + 3, RLENGTH); \
+			printf "  ${YELLOW}%-$(TARGET_MAX_CHAR_NUM)s${RESET} %s\n", command, message; \
+		} \
+	} \
+	{ lastLine = $$0 }' $(MAKEFILE_LIST)
+	@echo ''
+	@echo ''
+
+## Run project test cases
+test:
+	${INFO} "Building required docker images for testing"
+	@ echo " "
+	@ docker volume create --name=cache > /dev/null
+	@ echo " "
+	@ docker-compose -p $(DOCKER_TEST_PROJECT) -f $(DOCKER_TEST_COMPOSE_FILE) build --pull test
+	${SUCCESS} "Build Completed successfully"
+	@ echo " "
+	${INFO} "Running tests in docker container"
+	@ docker-compose -p $(DOCKER_TEST_PROJECT) -f $(DOCKER_TEST_COMPOSE_FILE) up test
+	${INFO} "Test results: exits if status code !=0"
+	${CHECK} $(DOCKER_TEST_PROJECT) $(DOCKER_TEST_COMPOSE_FILE) test
+	${INFO} "Copy coverage reports..."
+	@ docker cp $$(docker-compose -p $(DOCKER_TEST_PROJECT) -f $(DOCKER_TEST_COMPOSE_FILE) ps -q test):/application/coverage .
+	@ echo ""
+	@ ${INFO} "Cleaning workspace after test"
+	@ docker-compose -p $(DOCKER_TEST_PROJECT) -f $(DOCKER_TEST_COMPOSE_FILE) down -v
+	${INFO} "Removing dangling images..."
+	@ docker images -q -f dangling=true | xargs -I ARGS docker rmi -f ARGS
+	@ docker images -q -f dangling=true -f label=application=$(PROJECT_NAME) | xargs -I ARGS docker rmi -f ARGS
+	${SUCCESS} "Clean complete"
+
+
+## Delete local development server containers
+clean:
+	${INFO} "Cleaning your local environment, note all ephemeral volumes will be destroyed"
```

Comment:
The usage of this makefile should also be added to the readme

Reply:
Noted.